servicetalk-http-utils/src/main/java/io/servicetalk/http/utils/auth/BasicAuthHttpServiceFilter.java

```diff
@@ -141,16 +148,43 @@ public Builder(final CredentialsVerifier<UserInfo> credentialsVerifier,
         }
 
         /**
-         * Sets a {@link Key key} to store a user info object of authenticated user in {@link AsyncContextMap}.
+         * Sets a {@link AsyncContextMap.Key key} to store a user info object of authenticated user in
+         * {@link AsyncContext}.
          *
-         * @param userInfoKey a key to store a user info object in {@link AsyncContextMap}
+         * @param userInfoKey a key to store a user info object in {@link AsyncContext}
          * @return {@code this}
+         * @deprecated Use {@link #userInfoAsyncContextKey(ContextMap.Key)}
          */
-        public Builder<UserInfo> userInfoKey(final Key<UserInfo> userInfoKey) {
+        @Deprecated
+        public Builder<UserInfo> userInfoKey(final AsyncContextMap.Key<UserInfo> userInfoKey) {
             this.userInfoKey = requireNonNull(userInfoKey);
             return this;
         }
 
+        /**
+         * Sets a {@link ContextMap.Key key} to store a user info object of authenticated user in {@link AsyncContext}.
+         *
+         * @param userInfoAsyncContextKey a key to store a user info object in {@link AsyncContext}
+         * @return {@code this}
+         */
+        public Builder<UserInfo> userInfoAsyncContextKey(final ContextMap.Key<UserInfo> userInfoAsyncContextKey) {
```

Comment:
Consider offering a `holder` / `storage` API to instruct the filter which storage to use for the user-info instead of having storage specific APIs?

Reply:
This is an interesting idea... After thinking about a `BiConsumer<UserInfo, HttpRequestMetaData>` API, decided to keep as-is for the following reasons:
1. It's easier for users to migrate from the existing deprecated method. You just change the method name and a key type and you are done. No need to learn new API.
2. Harder to explain them options how users can use this consumer and where they can store the `UserInfo`.
3. Currently, it also requires to pass `HttpRequestMetaData` to let them put info into a request context. If later we add `ContextMap` feature for a connection, we will have to change API to take `ConnectionInfo` or provide a new API that targets `ConnectionInfo`.
4. No need to handle exceptions from the user-provided function/consumer.

If you have other ideas, please share.